src/agg/process_markers_symbolizer.cpp

```diff
@@ -128,16 +133,15 @@ struct agg_markers_renderer_context : markers_renderer_context
 
                     // Build local transformation matrix by resetting translation coordinates
                     agg::trans_affine marker_tr_copy(marker_tr);
-                    marker_tr_copy.tx = sample_x / sampling_rate - x0;
-                    marker_tr_copy.ty = sample_y / sampling_rate - y0;
+                    marker_tr_copy.tx = dx - x0;
+                    marker_tr_copy.ty = dy - y0;
 
                     // Create fill image
-                    std::shared_ptr<image_rgba8> fill_img;
```

Comment:
I think redefining this as `rgba8` was part of the optimization.

Reply:
I didn't change the definition, just changed places (both are and were `std::shared_ptr<image_rgba8>`).